### PR TITLE
[Typing][B-61] Add type annotations for `python/paddle/text/datasets/imdb.py`

### PR DESCRIPTION
**Pull request by enkilee**
### PR Category

User Experience

### PR Types

Improvements

### Description

- https://github.com/PaddlePaddle/Paddle/issues/65008
B61

### CODE REVIEW

File: python/paddle/text/datasets/imdb.py [hunk 1]
```diff
@@ -82,7 +90,21 @@ class Imdb(Dataset):
...
+    docs: list
+    labels: list
+
+    def __init__(
+        self,
+        data_file: str | None = None,
+        mode: _ImdbDataSetMode = 'train',
+        cutoff: int = 150,
+        download: bool = True,
+    ):
```
Comment (megemini) [inline python/paddle/text/datasets/imdb.py]:
```suggestion
    ) -> None:
```

File: python/paddle/text/datasets/imdb.py [hunk 2]
```diff
@@ -82,7 +90,21 @@ class Imdb(Dataset):
 
     """
 
-    def __init__(self, data_file=None, mode='train', cutoff=150, download=True):
+    data_file: str | None
+    mode: _ImdbDataSetMode
+    cutoff: int
+    download: bool
```
Comment (megemini) [inline python/paddle/text/datasets/imdb.py]:
```suggestion
```

貌似没有这两个属性？

File: python/paddle/text/datasets/imdb.py (hunk 1 above)
Comment (enkilee) [reply python/paddle/text/datasets/imdb.py]: 抱歉，忘了

File: python/paddle/text/datasets/imdb.py (hunk 2 above)
Comment (enkilee) [reply python/paddle/text/datasets/imdb.py]: 恩对。抱歉

File: python/paddle/text/datasets/imdb.py [hunk 3]
```diff
@@ -154,8 +174,10 @@ def _load_anno(self):
             self.docs.append([self.word_idx.get(w, UNK) for w in doc])
             self.labels.append(1)
 
-    def __getitem__(self, idx):
+    def __getitem__(
+        self, idx: int
+    ) -> tuple[npt.NDArray[np.int_], npt.NDArray[np.int_],]:
```
Comment (SigureMo) [inline python/paddle/text/datasets/imdb.py]:
```suggestion
    ) -> tuple[npt.NDArray[np.int_], npt.NDArray[np.int_]]:
```

File: python/paddle/text/datasets/imdb.py [hunk 4]
```diff
@@ -120,7 +140,7 @@ def _build_work_dict(self, cutoff):
         word_idx['<unk>'] = len(words)
         return word_idx
 
-    def _tokenize(self, pattern):
+    def _tokenize(self, pattern: Pattern) -> list[list[str]]:
```
Comment (SigureMo) [inline python/paddle/text/datasets/imdb.py]:
```suggestion
    def _tokenize(self, pattern: Pattern[str]) -> list[list[str]]:
```

**Review by SigureMo — APPROVED** [review]
LGTMeow

**Pull request merged**